### ISSUE TITLE
trackinghive.com seems down so your app does not work

### ISSUE THREAD

**Issue by sochartgit** [fireshaper/parcelpony#4]
Hi,
I've been very interested in your software but as it is linked to trackinghive.com
and it seems down, I am not able to use it anymore.
Any solution ?

**Comment by fireshaper** [comment]
Use https://my.trackinghive.com. It seems they do not have the main domain set up.

**Comment by sochartgit** [comment]
Thanks for your reply. I've tried this subdomain.
When I try to signup, nothing happens. I have disable any adblock.
![image](https://github.com/fireshaper/parcelpony/assets/83184203/8e42b420-a5ac-454a-a325-27cb8ec6c402)

**Comment by Master-Max** [comment]
Out of curiosity, what is trackinghive? Who is maintaining it? Does anyone know?

Going through the wayback machine it looks like it was made by some coders from Ranchi in 2016 but the about page has been missing since 2017...

Just wondering if anyone knows

**Comment by remmycat** [comment]
@Master-Max I was wondering the same thing, because offering an API like this for free without any info seems sketchy, or at best weird…

I'm not sure if the old captures are from the same domain owner.

I created an account out of curiosity, which seemed to work for me and leads to a dashboard where one can register their own subdomain and configure some things.

The "support" link leads to now-offline https://www.trackhive.co/contact-us/
But there the internet archive was more fruitful. While the contact site doesn't give out any information, the [privacy policy](https://web.archive.org/web/20230219222707/https://www.trackhive.co/privacy/) identifies them as `Trackhive (OpenXcell Technolabs Pvt. Ltd.)`. The site also defines terms of use and says the service is currently free but maybe won't be in the future.

The OpenXcell name then leads to this page that is still live today, about OpenXcell being hired to implement Trackhive, a platform targeted at eCommerce businesses: https://www.openxcell.com/our-work/trackhive/
This page has a link to a full case study PDF with some history, where the project is just called OrderHive https://www.openxcell.com/wp-content/uploads/2021/06/Case-study-Orderhive-PDF.pdf 

This also explains why the Privacy Policy also calls itself part of the "OrderHive Family" once. This seems to be a different name for the same thing, or a somehow related product, that was acquired by cin7 in 2021, and is available at https://my.orderhive.com/

So all in all this makes it seem like, maybe [cin7](https://cin7.com) are operating the site or API today, but I wouldn't know for sure. I would not bet on it staying online indefinitely.

**Comment by fireshaper** [comment]
@Master-Max I had reached out to the support team when I first found TrackingHive (Aug 2020) and asked about pricing to use the API. They told me at that time they were going to be working on pricing tiers but they had not implemented it. I was never given any more information than that though. The support agent's email address was from orderhive.com, like @remmycat mentioned.

It may end up that their product doesn't last and I'll have to find something else or abandon this project entirely.

**Comment by brandon-toner** [comment]
+1 to experiencing trackinghive outage. I'm getting the following in logs:

`Access to XMLHttpRequest at 'https://api.trackinghive.com/user/register' from origin 'https://my.trackinghive.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource`

Is there an alternate API or a workaround?

**Comment by Master-Max** [comment]
@brandon-toner I ended up requesting API access to the major US shippers for their tracking backend. 

The most annoying one to access was USPS, their visibility for their API is lacking. 
The Tracking V3 API is what I ended up using: [https://developer.usps.com/getting-started](url) 

I did have to contact them directly at: [mailto:APISUPPORT@usps.gov](url) in order to complete the form to get Enhanced Public Access II permission, they were very helpful but it did take 5 days to get full access.

FedEx, UPS, and DHL were much easier to access, though there does seem to be a thing about selling labels on your website if your project involves using their apis in a public way, using the api for internal use didn't have this requirement.

Additionally these rules are always subject to change, so apologies if my info is out of date.
